### PR TITLE
adding code for spawning a block

### PR DESCRIPTION
**Pull request by rads560**

### CODE REVIEW

File: Assets/Drag.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+public class Drag : MonoBehaviour
+{
+    ManipulationHandler manipulationHandler;
+    Vector3 startingPosition;
+    GameObject clone;
+
+    private void Awake()
+    {
+        //Debug.Log("running awake");
```
Comment (tgroechel) [inline Assets/Drag.cs]: Try not to push any commented code

File: Assets/Drag.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        startingPosition = transform.position;
+        manipulationHandler = GetComponent<ManipulationHandler>();
+        manipulationHandler.OnManipulationStarted.AddListener(startedMotion);
+        manipulationHandler.OnManipulationEnded.AddListener(stoppedMotion);
+    }
+
+    private void stoppedMotion(ManipulationEventData arg0)
+    {
+        if (Vector3.Distance(transform.position, startingPosition) < 2) // how far you can move the block while it still will snap back
```
Comment (tgroechel) [inline Assets/Drag.cs]: This should really be done with Unity events (https://www.youtube.com/watch?v=mC_yUEnGQXM) , you should be able to see `OnManipulationEnded` if the block is still in collision with the other or not

File: Assets/Drag.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            Destroy(this.gameObject);
```
Comment (tgroechel) [inline Assets/Drag.cs]: Delete `this` , you can just write `gameObject`, same thing goes for `transform`, it is actually poor practice to write `this`

File: Assets/Drag.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+    }
+
+    public void startedMotion(ManipulationEventData arg0)
+    {
+        //Debug.Log("started motion");
```
Comment (tgroechel) [inline Assets/Drag.cs]: Same thing here with commented out pushes

File: Assets/Drag.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        if (Vector3.Distance(transform.position, startingPosition) < 1)
+        {
+            clone = Instantiate(this.gameObject, startingPosition, Quaternion.identity); //  make a clone in original position
```
Comment (tgroechel) [inline Assets/Drag.cs]: same "this" comment

File: Assets/Drag.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    private void Awake()
+    {
+        //Debug.Log("running awake");
+        startingPosition = transform.position;
+        manipulationHandler = GetComponent<ManipulationHandler>();
+        manipulationHandler.OnManipulationStarted.AddListener(startedMotion);
+        manipulationHandler.OnManipulationEnded.AddListener(stoppedMotion);
+    }
+
+    private void stoppedMotion(ManipulationEventData arg0)
```
Comment (tgroechel) [inline Assets/Drag.cs]: Functions/methods in C# are camel case but actually start with a capital letter (I know, bizarre) but just look at `GetComponent<>()` so make sure to write them camel starting with capital

**Review by tgroechel — CHANGES_REQUESTED** [review]
Looks pretty good to start just some simple fixes and changing to events

File: Assets/Drag.cs (hunk 3 above)
Comment (rads560) [reply Assets/Drag.cs]: So I don't have to write transform.position either?

**Pull request closed without merging**